src/transformers/feature_extraction_sequence_utils.py

```diff
@@ -171,11 +171,6 @@ def pad(
                     "Should be one of a python, numpy, pytorch or tensorflow object."
                 )
 
-        for key, value in processed_features.items():
```

Comment:
Note that this is a breaking change. Previously, the user was allowed to pass lists of ints/floats:
```python
from transformers import AutoProcessor

fake_audio_sr = 16000
fake_audio_duration_in_seconds = 4.55
fake_audio_len = int(fake_audio_sr*fake_audio_duration_in_seconds)
fake_audio = fake_audio_len * [1.0]

audio_processor = AutoProcessor.from_pretrained("facebook/hubert-large-ls960-ft")
audio_processor(
    fake_audio,
    sampling_rate=fake_audio_sr,
    return_tensors="pt",
    padding=True,
)
```

However, this PR changes this behaviour by removing the cast to numpy:
```python
File ~/transformers/src/transformers/feature_extraction_sequence_utils.py:215, in SequenceFeatureExtractor.pad(self, processed_features, padding, max_length, truncation, pad_to_multiple_of, return_attention_mask, return_tensors)
    213 if key not in batch_outputs:
    214     batch_outputs[key] = []
--> 215 if value.dtype is np.dtype(np.float64):
    216     value = value.astype(np.float32)
    217 batch_outputs[key].append(value)

AttributeError: 'list' object has no attribute 'dtype'
```

We can be a bit more strategic in only casting lists of ints/floats to numpy, and leaving torch tensors and numpy arrays as they are